Fix tests

JanCaha opened the pull request:
Fixes #110 

REQUIRES MERGIN-CLIENT that is not yet released !!! Uses the new `delete_project_now()` function.

add test for init from db

fixes some minor issues in tests from previous versions

wonder-sk commented on test/conftest.py:
```diff
@@ -205,3 +188,127 @@ def mc():
+            "base": db_schema_base,
+            "mergin_project": full_project_name,
+            "sync_file": filename_sync_gpkg(),
+        }
```
looks like unnecessary code duplication... how about this:
```python
connection = { ... }
if ignored_tables:
  connection[skip_tables] = ignored_tables
```

wonder-sk commented on test/test_init_db.py:
```diff
@@ -0,0 +1,207 @@
+)
+
+
+def test_init_from_db(mc: MerginClient, db_connection):
```
can you please add a brief docstring about what each test tries to do?

wonder-sk commented on test/test_init_db.py:
```diff
@@ -0,0 +1,207 @@
+            "CONNECTIONS": [connection],
+            "init_from": "db",
+        }
+    )
```
this looks like a copy of init_sync_from_db() - to reduce code duplication I would suggest to reuse init_sync_from_db(), and make it accept any SQL file to populate the DB (just like init_sync_from_geopackage() gets source GPKG file), then add another .sql file in addition to create_base.sql that would populate a different schema...

wonder-sk commented on test/conftest.py:
```diff
@@ -205,3 +188,127 @@ def mc():
+        base_table_dump = file.read()
+
+    base_table_dump = base_table_dump.replace("default-schema-name", db_schema_main)
+    base_table_dump = base_table_dump.replace("default-table-name", "simple")
```
I would prefer not to do this kind of replacement, and simply have .sql files with the expected schema/table names hardcoded. I think we do not need to have a different schema for each test - in theory all the tests could use the same "main" schema name in DB, they are run sequentially so it should not hurt, and it would simplify things (e.g. one could also copy&paste the commands from .sql file to a sql console without modifications)

wonder-sk commented on test/test_basic.py:
```diff
@@ -638,6 +638,7 @@ def test_with_local_changes(
     dbsync_status(mc)
 
 
+@pytest.mark.skip(reason="Currently being reworked in MM")
```
why does this need to be marked as skipped?

JanCaha replied:
The test was failing when runned localy, I thought it might be caused by changes in mergin-client. But it seems to be working now, so maybe it was related to project marked for deletion from previous test runs... anyways the skip is removed.

JanCaha commented:
Ok, this seems to be ready for being merge:

- CI now installs according to requirements.txt
- version of mergin-client raised to 0.9.0
- fixed some deprecated calls in db-sync (mostly use of MerginProject.metadata)
- in function `_get_mergin_project()` we need to strictly reread project metadata for cached projects, otherwise old metadata pop up, even if the `mergin.json` file has newer info
- fixed some tests
- added tests for init_from_db
